### PR TITLE
issue #4: add hit highlights

### LINKED ISSUE #4: Implement Azure Hit Highlighting in Semantic Search in azure-db
Integrate Azure's hit highlighting to improve user experience and relevance in our semantic search system.

### Rationale
- Facilitates quicker identification of relevant information by visually marking matching search terms.
- Highlights and focuses on the most pertinent parts of texts related to the search query instead of returning the whole content.

### Tasks
- Implement Azure's hit highlighting into our search results.
- Configurable highlighting style.
- Tests.

### Acceptance Criteria
- Search results include relevant snippets.

### PR DESCRIPTION
**Pull request by k-allagbe**
Closes #4: hit highlighting

- return snippet instead of whole content
- highlight the used keywords
- configurable html highlighting style

### CODE REVIEW

File: azure-ai-search/tests/test_azure_search.py [hunk 1]
```diff
@@ -38,13 +37,35 @@ def test_transform_result(self):
...
+            "@search.score": 1.23,
+            "metadata_storage_name": "Example Title",
+            "@search.highlights": None,
+            "subtitle": "Example Subtitle",
+            "metadata_last_modified": "2023-01-01T00:00:00Z",
+        }
+        expected_output_no_highlights = {
+            "id": "123",
+            "url": "http://example.com",
+            "score": 1.23,
```
Comment (rngadam) [inline azure-ai-search/tests/test_azure_search.py:60]: are search score coming from AI Search not normalized between 0 and 1?

**Review by rngadam — CHANGES_REQUESTED** [review]

File: azure-ai-search/tests/test_azure_search.py (hunk 1 above)
Comment (k-allagbe) [reply azure-ai-search/tests/test_azure_search.py:61]: I believe they are unbounded. Check https://learn.microsoft.com/en-us/azure/search/index-similarity-and-scoring#scoring-algorithms-used-in-full-text-search

File: azure-ai-search/index_search/__init__.py [hunk 2]
```diff
@@ -36,8 +70,14 @@ def search(query, config: AzureIndexSearchConfig):
...
-        search_results = config.client.search(search_text=query)
+        search_results = config.client.search(
+            search_text=query,
+            highlight_fields=config.highlight_fields,
+            highlight_pre_tag=f"<{config.highlight_tag}>",
+            highlight_post_tag=f"</{config.highlight_tag}>",
+        )
         transformed_results = [transform_result(result) for result in search_results]
         return transformed_results
     except Exception as e:
```
[the remark this answers is not in the document]
Comment (k-allagbe) [reply azure-ai-search/index_search/__init__.py]: Done. I used AzureError for now.

File: azure-ai-search/tests/test_azure_search.py (hunk 1 above)
Comment (rngadam) [reply azure-ai-search/tests/test_azure_search.py:61]: ok, but then within the particular set of results you return you could compute doc_score/(max(doc_score) - min(doc_score))?

we have to think also about integrating the other signals in the postgresql database to rerank the results.

Comment (rngadam) [reply azure-ai-search/tests/test_azure_search.py:61]: (probably a separate issue to work on this)

File: azure-ai-search/index_search/__init__.py [hunk 3]
```diff
@@ -14,19 +15,53 @@ class EmptyQueryError(Exception):
 
 @dataclass
 class AzureIndexSearchConfig:
-    endpoint: str
-    api_key: str
-    index_name: str
+    """
+    Configuration class for Azure Index Search.
+
+    This class holds the necessary configuration details required for setting up
```
Comment (rngadam) [inline azure-ai-search/index_search/__init__.py:21]: this paragraph doesn't seem to wrap on 80

**Review by rngadam — CHANGES_REQUESTED** [review]
small nits and one possibly postpone piece of work

File: azure-ai-search/tests/test_azure_search.py (hunk 1 above)
Comment (k-allagbe) [reply azure-ai-search/tests/test_azure_search.py:61]: Here: https://github.com/ai-cfia/azure-db/issues/8

File: azure-ai-search/index_search/__init__.py (hunk 3 above)
Comment (k-allagbe) [reply azure-ai-search/index_search/__init__.py]: Ruff allows up to 88 chars per line inside the code. Could be configured to 80. The description is also too verbose imo so I shortened it.

**Review by rngadam — APPROVED** [review]

**Pull request merged**